burnman/data/input_raw_endmember_datasets/HPdata_to_burnman.py

```diff
@@ -74,74 +68,105 @@ def getmbr(ds, mbr):
-        if M.flag == 1:
-            print '            \'landau_hp\', {\'P_0\':', 1.e5, ','
-            print '                          \'T_0\':', 298.15, ','
-            print '                          \'Tc_0\':', M.od[0], ','
-            print '                          \'S_D\':', M.od[1] * 1e3, ','
-            print '                          \'V_D\':', M.od[2] * 1e-5, '}]]'
-        if M.flag == 2:
-            print '            \'bragg_williams\', {\'deltaH\':', M.od[0] * 1e3, ','
-            print '                               \'deltaV\':', M.od[1] * 1e-5, ','
-            print '                               \'Wh\':', M.od[2] * 1e3, ','
-            print '                               \'Wv\':', M.od[3] * 1e-5, ','
-            print '                               \'n\':', M.od[4], ','
-            print '                               \'factor\':', M.od[5], '}]]'
-        print '        Mineral.__init__(self)'
-        print ''
-print ''
+with open('HP_2011_ds62.py', 'wb') as outfile:
+    outfile.write('# This file is part of BurnMan - a thermoelastic and '
+                  'thermodynamic toolkit for the Earth and Planetary Sciences\n'
+                  '# Copyright (C) 2012 - 2018 by the BurnMan team, '
+                  'released under the GNU \n# GPL v2 or later.\n\n\n'
+                  '"""\n'
+                  'HP_2011 (ds-62)\n'
+                  'Endmember minerals from Holland and Powell 2011 and references therein\n'
+                  'Update to dataset version 6.2\n'
+                  'The values in this document are all in S.I. units,\n'
+                  'unlike those in the original tc-ds62.txt\n'
+                  'File autogenerated using HPdata_to_burnman.py\n'
+                  '"""\n\n'
+                  'from ..mineral import Mineral\n'
+                  'from ..processchemistry import dictionarize_formula, formula_mass\n\n')
+
+    outfile.write('"""\n'
+                  'ENDMEMBERS\n'
+                  '"""\n\n')
+    
+    formula = '0'
+    for i in range(int(ds[0][0])):
+        mbr = ds[i * 4 + 3][0]
+        M = getmbr(ds, mbr)
+        if mbr == 'and':  # change silly abbreviation
+            mbr = 'andalusite'
+        if M.flag != -1 and M.flag != -2 and M.k[0] > 0:
+            outfile.write('class {0} (Mineral):\n'.format(mbr)+
+                          '    def __init__(self):\n'
+                          '        formula = \'{0}\''.format(M.formula)+'\n'
+                          '        formula = dictionarize_formula(formula)\n'
+                          '        self.params = {{\n'
+                          '            \'name\': \'{0}\','.format(M.name)+'\n'
+                          '            \'formula\': formula,\n'
+                          '            \'equation_of_state\': \'hp_tmt\',\n'
+                          '            \'H_0\': {0},'.format(M.H * 1e3)+'\n'
+                          '            \'S_0\': {0},'.format(M.S * 1e3)+'\n'
```

Comment:
this also looks more complicated than it should. Can you not construct the dictionary natively and pretty-print it?

Reply:
I don't know whether there's an easy way to get the indenting correct in the output code. This was the neatest solution I could find without pickling (which can't deal with classes anyway).